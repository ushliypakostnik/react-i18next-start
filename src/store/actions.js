import CONTENT from './content';

// Actions Types

export const AUTH_REQUEST = 'AUTH_REQUEST';
export const AUTH_SUCCESS = 'AUTH_SUCCESS';
export const AUTH_ERROR = 'AUTH_ERROR';
export const AUTH_LOGOUT = 'AUTH_LOGOUT';

export const NEWS_REQUEST = 'NEWS_REQUEST';
export const NEWS_SUCCESS = 'NEWS_SUCCESS';
export const NEWS_ERROR = 'NEWS_ERROR';

// Action Creators

export const authRequest = () => ({
  type: AUTH_REQUEST
});

export const authSuccess = () => ({
  type: AUTH_SUCCESS
});

export const authError = (error) => ({
  type: AUTH_ERROR,
  error
});

// fake async fetch
function authResult(dispatch) {
  dispatch(authSuccess());
}
export const fetchAuth = (credentials) => {
  return dispatch => {
    dispatch(authRequest(credentials));
    setTimeout(authResult(dispatch), 100);
  }
}

export const authLogout = () => ({
  type: AUTH_LOGOUT
});

export const newsRequest = () => ({
  type: NEWS_REQUEST
});

export const newsSuccess = (news) => ({
  type: NEWS_SUCCESS,
  news
});

export const newsError = (error) => ({
  type: NEWS_ERROR,
  error
});

// fake async fetch
function newsResult(dispatch) {
  dispatch(newsSuccess(CONTENT));
}
export const fetchNews = () => {
  return dispatch => {
    dispatch(newsRequest());
    setTimeout(newsResult(dispatch), 100);
  }
};
