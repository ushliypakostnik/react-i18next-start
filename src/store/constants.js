import Cookies from "js-cookie";

export const COOKIES = {
  LANG: {
    name: 'language',
    expires: 7,
  },
};

const isProd = process.env.NODE_ENV === 'production';
const apiUrl = process.env.REACT_APP_API_URL;
export const API_URL = isProd ? apiUrl || 'https://...' : apiUrl || 'http://localhost:8082';

export const LANGUAGES = [
  { id: 1, name: 'en' },
  { id: 2, name: 'ru' },
];

// Auto language
const language = Cookies.get(COOKIES.LANG.name) || null;
export const AUTO_LANG = language || LANGUAGES[1].name;

export const INITIAL_STATE = {
  rootReducer: {
    utils: {
      language: AUTO_LANG,
    },
    reducer2: {

    },
  },
};

export const LOCALSTORAGE = {
  ITEM: 'item',
}
