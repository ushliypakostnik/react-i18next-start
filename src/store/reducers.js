import { combineReducers } from 'redux';

import utils from './modules/utils/reducer';
import reducer2 from './modules/module2/reducer';

const rootReducer = combineReducers({
  utils,
  reducer2,
});

export default rootReducer;
