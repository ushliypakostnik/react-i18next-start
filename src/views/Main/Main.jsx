import React, { Fragment } from "react";
import { useTranslation } from 'react-i18next';

import '../../scss/views/_page.scss';

const Main = () => {
  const { t } = useTranslation();

  return (
    <Fragment>
      <section className="page--main">
        <div className="container">
          <h2>{t('views.main.title')}</h2>
        </div>
      </section>
    </Fragment>
  );
};

export default Main;
