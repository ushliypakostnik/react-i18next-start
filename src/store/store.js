import { createStore, combineReducers, applyMiddleware } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import { connectRouter } from 'connected-react-router';
import { createBrowserHistory as createHistory } from 'history';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';

// eslint-disable-next-line no-unused-vars
import { INITIAL_STATE, LOCALSTORAGE } from './constants';
import rootReducer from './reducers';

const middlewares = [];
middlewares.push(thunkMiddleware)

if (process.env.NODE_ENV !== 'production') {
  const loggerMiddleware = createLogger();

  middlewares.push(loggerMiddleware);
}

const localStorageMiddleware = ({getState}) => {
  return (next) => (action) => {
    const result = next(action);
    /*
    if (getState().rootReducer.module.boolean) {
      localStorage.setItem(LOCALSTORAGE.ITEM, JSON.stringify(
          getState().rootReducer.module.field,
      ));
    }
    */
    return result;
  };
};
middlewares.push(localStorageMiddleware);

const reHydrateStore = (state) => {
  /*
  if (localStorage.getItem(LOCALSTORAGE.ITEM) !== null) {
    const localData = JSON.parse(localStorage.getItem(LOCALSTORAGE.ITEM) || '{}');
    const _state = Object.assign({}, state, {
      rootReducer: {
        ...state.rootReducer,
        user: {
          ...state.rootReducer.module,
          field: localData,
        },
      },
    });
    return _state;
  }
  */
  return state;
};

export const history = createHistory();
middlewares.push(routerMiddleware(history));

function configureStore(state) {
  return createStore(
    combineReducers({
      rootReducer,
      router: connectRouter(history),
    }),
    reHydrateStore(state),
    applyMiddleware(...middlewares)
  );
}

const store = configureStore(INITIAL_STATE);

// console.log('Store: ', store.getState());

export default store;
