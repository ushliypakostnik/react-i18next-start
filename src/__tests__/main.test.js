import React from 'react';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { INITIAL_STATE } from '../store/constants';

import Main from '../containers/main';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('app render', () => {
  const props = {};
  const store = mockStore(INITIAL_STATE);

  const wrapper = mount(<Provider store={store}>
                          <MemoryRouter>
                            <Main {...props} />
                          </MemoryRouter>
                        </Provider>);

  it('app render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('app contain wrapper with class', () => {
    expect(wrapper.find('.message')).toHaveLength(1);
  });

  it('app componentDidMount', () => {
    const spy = jest.spyOn(wrapper.find('Main').instance(), 'componentDidMount');
    wrapper.find('Main').instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
});
