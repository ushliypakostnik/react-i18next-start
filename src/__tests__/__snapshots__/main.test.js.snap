// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app render app render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "0u7rn1",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "0u7rn1",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withI18nextTranslation(Connect(Main))>
        <Connect(Main)
          i18n={Object {}}
          t={[Function]}
          tReady={false}
        >
          <Main
            i18n={Object {}}
            language="ru"
            setLanguage={[Function]}
            t={[Function]}
            tReady={false}
          >
            <div
              className="page-center"
            >
              <div
                className="message"
              >
                <h3>
                  title
                </h3>
                <button
                  key="0"
                  onClick={[Function]}
                >
                  en
                </button>
                <button
                  key="1"
                  onClick={[Function]}
                >
                  ru
                </button>
              </div>
            </div>
          </Main>
        </Connect(Main)>
      </withI18nextTranslation(Connect(Main))>
    </Router>
  </MemoryRouter>
</Provider>
`;
